docs/component-stability.md

```diff
@@ -66,6 +66,99 @@ Stable components MUST be compatible between minor versions unless critical secu
+- Of course, a component may define additional internal telemetry which is not in this list.
+
+- Some of this internal telemetry may already be provided by pipeline auto-instrumentation or
+helper modules (such as `receiverhelper`, `scraperhelper`, `processorhelper`, or `exporterhelper`).
+Please check the documentation to verify which parts, if any, need to be implemented manually.
+
+**Definition:** In the following, an "item" refers generically to a single log record, metric point,
+or span.
+
+The internal telemetry of a stable pipeline component should allow observing the following:
+
+1. How much data the component receives.
+
+    For receivers, this could be a metric counting requests, received bytes, scraping attempts, etc.
+
+    For other components, this would typically be the number of items received through the
+    `Consumer` API.
+
+2. How much data the component outputs.
+
+    For exporters, this could be a metric counting requests, sent bytes, etc.
+
+    For other components, this would typically be the number of items forwarded to the next
+    component through the `Consumer` API.
+
+3. How much data is dropped because of errors.
+
+    For receivers, this could include a metric counting payloads that could not be parsed in.
+
+    For receivers and exporters that interact with an external service, this could include a metric
+    counting requests that failed because of network errors.
+
+    The goal is to be able to easily pinpoint the source of data loss in the Collector pipeline, so
+    this should either:
+    - only include errors internal to the component, or;
+    - allow distinguishing said errors from ones originating in an external service, or propagated
+        from downstream Collector components.
+
+4. Details for error conditions.
+
+    This could be in the form of logs or spans detailing the reason for an error. As much detail as
+    necessary should be provided to ease debugging. Processed signal data should not be included for
+    security and privacy reasons.
+
+5. Other possible discrepancies between input and output, if any. This may include:
+
+    - How much data is dropped as part of normal operation (eg. filtered out).
+
+    - How much data is created by the component.
+
+    - How much data is currently held by the component (eg. an UpDownCounter keeping track of the
+        size of an internal queue).
+
+6. Processing performance.
+
+    This could include spans for each operation of the component, or a histogram of end-to-end
+    component latency.
+    
+    The goal is to be able to easily pinpoint the source of latency in the Collector pipeline, so
+    this should either:
+    - only include time spent processing inside the component, or;
+    - allow distinguishing this latency from that caused by an external service, or from time spent
+        in downstream Collector components.
+
+When measuring amounts of data, counting items is recommended. Where this can't easily be done, any
```

Comment:
are there any metrics where we should explicitly recommend an updowncounter/gauge?

Reply:
"Counting" in this sentence was not meant to recommend using a Counter, it was only meant to specify that we want to measure in units of "items" rather than bytes or something else. Since it doesn't seem to be clear, I'll reword this sentence.

Notably, "how much data is *currently* being held by the component" could very naturally be implemented with an UpDownCounter. I'll add it explicitly as an example implementation.